subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -55,9 +55,10 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include <errno.h>
 #include <zephyr/init.h>
 #include <zephyr/sys/printk.h>
-
+#include <net/socket.h>
```

Comment:
```suggestion
#include <zephyr/net/socket.h>
```

Is it really needed though here?

Reply:
OK I can remove and test.